sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -98,7 +98,8 @@ case class LocalLimitExec(limit: Int, child: SparkPlan) extends UnaryExecNode wi
 /**
  * Take the `limit` elements of the child output.
  */
-case class GlobalLimitExec(limit: Int, child: SparkPlan) extends UnaryExecNode {
+case class GlobalLimitExec(limit: Int, child: SparkPlan,
+                           orderedLimit: Boolean = false) extends UnaryExecNode {
```

Comment:
what does orderedLimit mean here?

Reply:
It means this global limit won't change input data order. This is used on sort + limit case which is usually taken by `TakeOrderedAndProjectExec` at most of time.

But if limit number is more than `SQLConf.TOP_K_SORT_FALLBACK_THRESHOLD`, it's not goes for `TakeOrderedAndProjectExec`.

Reply:
what do you mean by "it's not goes for TakeOrderedAndProjectExec"?

Reply:
When limit number is more than `SQLConf.TOP_K_SORT_FALLBACK_THRESHOLD`, the planner won't choose `TakeOrderedAndProjectExec` to perform the sort + limit operation.

Reply:
please document it in code.

Reply:
Ok, I see. I will document it in the pr.

Reply:
code needs to be documented. we won't find this pr discussion a year from now by looking at the source code, trying to figure out what it means. also the doc needs to be readable. the current doc for the config flag is unfortunately unparsable.

Reply:
I see. Let me submit a pr later to address those document. Really appreciate your comment.

Reply:
thanks @viirya 

can you write a design doc or put it in the classdoc of limit on how we handle limits? your sequence of prs are making limits much more complicated (with optimizations) and very difficult to reason about. i think we can make it easier to reason about, if we actually document the execution strategy.

Reply:
Ok. I will do it too.